### DISCUSSION TITLE
Metadata integrity cannot be verified, repository is disabled now

### DISCUSSION THREAD

**Discussion by Andinitri** [termux/termux-packages#19953] [Q&A]
I already changed the repo after wrote termux-change-repo. But the result is still same. Please tell me the solution. Thanks

**Comment by sylirre** [comment]
You are not explaining what you did. You are not providing a full context of the issue. Post-scriptum "Metadata integrity cannot be verified, repository is disabled now" is meaningless without context.

Integrity issue usually means you did something weird, such as decided to copy-paste commands from some tutorial that added third party repository. Of course third party repositories would not pass verification unless key was set up.

Also if you are trying to use Termux in VMOS, F1VM or similar applications, then most likely you won't be able to install packages. Such environments are not supported because they are full of system bugs.

**Reply by Andinitri** [reply]
I just downloaded the termux app and wrote apt update&& apt upgrade and the result like that sir. I dont know why and I should do what

**Reply by twaik** [reply]
***Termux on Google Play Store is OUTDATED*** 🪦
Termux and its plugins are no longer updated on the Google Play Store. This is due to API requirements that introduce new restrictions, which break Termux's basic functionality.
Please switch to the F-Droid version!
Direct link: <https://f-droid.org/en/packages/com.termux>
**Before** you install it, You **must** __fully uninstall__ current Termux and all plugins. If you don't, you will get a cryptic `app not installed` error!
(that means uninstalling the main app, as well as Termux:API, :Styling, :Widget, :Task, :Float, :Boot)
If you need help with backing up your files:
<https://wiki.termux.com/wiki/Backing_up_Termux>

**Reply by twaik** [reply]
After installing Termux and opening it, you should do the following:

**1.** `termux-change-repo` (to select a nearby mirror to get updates from):
- `SPACE` toggles highlighted options;
- `ENTER` moves to the next screen;
Choose the first/topmost option on each screen. _(If that one fails, try another.)_
**2.** `pkg update && pkg upgrade` (to update your package database, then upgrade all packages)
On each prompt, press `y` and `Enter` to install newest versions available
**3.** `exit` (to exit Termux, so you can restart it yourself, all updated!)

**Reply by Andinitri** [reply]
ohh okayy i understand thankss sir
